src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -383,249 +507,339 @@ private void TrainModels(Float[][] columnData, int[] rowCounts, IChannel ch)
                         Mkl.Gemm(Layout, Mkl.Transpose.NoTrans, Mkl.Transpose.NoTrans,
-                            ccol, ccol, ccol, 1, u, ccol, uInvScaled, ccol, 0, InvModels[iinfo], ccol);
+                            ccol, ccol, ccol, 1, u, ccol, uInvScaled, ccol, 0, _invModels[iinfo], ccol);
                     }
                 }
                 else
                     ch.Assert(false);
             }
         }
 
-        private WhiteningTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, TestColumn)
-        {
-            Host.AssertValue(ctx);
-
-            // *** Binary format ***
-            // <prefix handled in static Create method>
-            // <base>
-            // foreach added column
-            //   ColInfoEx
-            // foreach model
-            //   whitening matrix
-            //   recovery matrix
-
-            Host.AssertNonEmpty(Infos);
-            _exes = new ColInfoEx[Infos.Length];
-            for (int i = 0; i < _exes.Length; i++)
-                _exes[i] = new ColInfoEx(ctx, Infos[i]);
-
-            _models = new Float[Infos.Length][];
-            InvModels = new Float[Infos.Length][];
-            for (int i = 0; i < Infos.Length; i++)
-            {
-                _models[i] = ctx.Reader.ReadFloatArray();
-                ValidateModel(Host, _models[i], Infos[i].TypeSrc);
-                if (_exes[i].SaveInv)
-                {
-                    InvModels[i] = ctx.Reader.ReadFloatArray();
-                    ValidateModel(Host, InvModels[i], Infos[i].TypeSrc);
-                }
-            }
-            Metadata.Seal();
-        }
-
-        public static WhiteningTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
-        {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-            h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
-            ctx.CheckAtModel(GetVersionInfo());
-
-            // *** Binary format ***
-            // int: sizeof(Float)
-            // <remainder handled in ctors>
-            int cbFloat = ctx.Reader.ReadInt32();
-            h.CheckDecode(cbFloat == sizeof(Float));
-            return h.Apply("Loading Model", ch => new WhiteningTransform(h, ctx, input));
-        }
-
         public override void Save(ModelSaveContext ctx)
         {
             Host.CheckValue(ctx, nameof(ctx));
             ctx.CheckAtModel();
             ctx.SetVersionInfo(GetVersionInfo());
 
             // *** Binary format ***
-            // int: sizeof(Float)
+            // int: sizeof(float)
             // <base>
             // foreach added column
             //   ColInfoEx
             // foreach model
             //   whitening matrix
             //   recovery matrix
-            ctx.Writer.Write(sizeof(Float));
 
-            SaveBase(ctx);
+            SaveColumns(ctx);
```

Comment:
>SaveColumns [](start = 12, length = 11)

Does saving the columns here change the format for the transform? If so, we should rev the version info and have a test that loads the old version

Reply:
It does not! It's the general trend for transforms deriving from OneToOneTransformerBase.

Reply:
I follow you that this is general practice when deriving from OneToOne -- but now we are saving out column information, ColInfoEx, matrix, inv matrix.  where previously it looks to be only ColInfoEx, matrix, inv matrix. Or am I missing something? 

---
In reply to: [229557290](https://github.com/dotnet/machinelearning/pull/1452#discussion_r229557290) [](ancestors = 229557290)